src/main/java/seedu/clinic/logic/parser/FindCommandParser.java

```diff
@@ -25,9 +27,21 @@ public FindCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        String[] productNameKeywords = trimmedArgs.split("\\s+");
+        String type = productNameKeywords[0].toLowerCase();
 
-        return new FindCommand(new NameContainsKeywordsPredicateForSupplier(Arrays.asList(nameKeywords)));
+        // Ensures that the user enters the type and at least 1 keyword
+        if (!(type.equals("supplier") || type.equals("warehouse")) || productNameKeywords.length < 2) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        List<String> keywords = Arrays.asList(productNameKeywords);
+        if (type.equals("supplier")) {
+            return new FindCommand(new SupplierProductsContainKeywordsPredicate(keywords), null);
+        } else {
+            return new FindCommand(null, new WarehouseProductsContainKeywordsPredicate(keywords));
+        }
```

Comment:
Similar to the idea above, the presence of null here may no be so desirable.
Perhaps we can consider reduce the use of null here in future iterations?